### ISSUE TITLE
Dropdown buttons accessibility issues

### ISSUE THREAD

**Issue by jamesjacobs** [jadu/pulsar#985]
There are various a11y issues with the markup generated by `html.button_dropdown` and `html.actions_menu`, these include:

1. Lack of `aria-haspopup="true"` which notifies screen reader users that the button does something different than submitting a form etc.
2. Lack of `aria-expanded="false/true"` which notifies screen reader users if the button is expanded or not.
3. Potential use of `aria-controls` tied in to the menu the button reveals
4. Typical keyboard controls associated with menus
5. Pressing ESC should close the menu
6. Focus management, focus should shift the the `ul` which would need an accessible name`. Should focus should be trapped within the menu? Needs research.
7. ~We should also explore the use of the following roles, `menu`, `menuitem` and `none` on the `li`~ Nope, menuitems should not have interactive descendants - https://w3c.github.io/html-aria/#index-aria-menuitem 

See: 
https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html
